src/Admin/test/VMware.ScriptRuntimeService.AdminApi.Tests/Authentication/Basic/BasicAuthenticationHandlerTests.cs

```diff
@@ -116,7 +116,29 @@ internal class BasicAuthenticationHandlerTests {
          Assert.IsFalse(result.Succeeded);
          Assert.AreEqual("Invalid username or password", result.Failure.Message);
       }
-      
+
+      [Test]
+      public async Task HandleAuthenticateAsync_UseSaltedPasswordDirectly_ReturnsAuthenticateResultSuccess() {
+         // Arrange
+         var username = "administrator@vsphere.local";
+         var password = "Admin!23";
+         var salt = "g9H6g+AGZOY/uA8+";
+         var saltedPassword = GetSha256Hash(salt + password);
+         var header = FormatAuthorizationHeader(username, saltedPassword);
+
+         _environment.Reset();
+         _environment.Setup(e => e.GetEnvironmentVariable("ADMIN_USER")).Returns(username);
+         _environment.Setup(e => e.GetEnvironmentVariable("ADMIN_PASSWORD")).Returns(saltedPassword);
+         _environment.Setup(e => e.GetEnvironmentVariable("ADMIN_PASSWORD_SALT")).Returns(salt);
+
+         // Act
+         var result = await AuthenticateAsync(header);
+
+         // Assert
+         Assert.IsFalse(result.Succeeded);
+         Assert.AreEqual("Invalid username or password", result.Failure.Message);
+      }
+
       [Test]
```

Comment:
Add a test where `var password = "g9H6g+AGZOY/uA8+Admin!23";` and `var salt = "g9H6g+AGZOY/uA8+"`